examples/pytorch/rgcn/entity_classify.py

```diff
@@ -151,6 +156,8 @@ def main(args):
             help="l2 norm coef")
     parser.add_argument("--relabel", default=False, action='store_true',
             help="remove untouched nodes and relabel")
+    parser.add_argument("--use-self-loop", default=False, action='store_true',
+            help="include self feature as a special relation")
```

Comment:
Actually, I don't think self loop is a special relation. It works more like skip connection especially because self loop is not done in message passing and is not normalized.

Even if it is a special relation, the preprocessing code of entity classification already adds a self-relation to the graph. I agree this argument makes the usage layer more consistent. But if user sets this argument to be true, then self loop will then be added twice. So maybe we should remove the code in data preprocessing that adds the self relation.

Reply:
It's more like a special relation because of the W_0 weight matrix. By contrast, GraphSAGE models it as $W * (mean({h_j}) + h_i)$ which is more like a skip connection. Does the author's code add self loops?

Reply:
Skip connection usually involves a transformation matrix $W$ to make sure dimension matches. 

Author's code adds an identity matrix here:
https://github.com/tkipf/relational-gcn/blob/master/rgcn/prepare_dataset.py#L39